api/src/main/java/org/infinispan/api/async/Lock.java

```diff
@@ -0,0 +1,20 @@
+package org.infinispan.api.async;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public interface Lock {
+   String name();
+
+   CompletableFuture<Void> lock();
```

Comment:
These should be `CompletionStage`. In fact we shouldn't have `CompletableFuture` anywhere in the API.

Reply:
Fixed.